internal/garbageCollector/garbageCollector.go

```diff
@@ -0,0 +1,85 @@
+package garbageCollector
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+	clusterPkg "github.com/openshift/assisted-service/internal/cluster"
+	"github.com/openshift/assisted-service/internal/host"
+	"github.com/openshift/assisted-service/pkg/leader"
+	"github.com/openshift/assisted-service/pkg/s3wrapper"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+	"github.com/sirupsen/logrus"
+)
+
+type Config struct {
+	DeletedUnregisteredAfter time.Duration `envconfig:"DELETED_UNREGISTERED_AFTER" default:"168h"`
+	DeregisterInactiveAfter  time.Duration `envconfig:"DELETED_INACTIVE_AFTER" default:"720h"`
+}
+
+type GarbageCollectors interface {
+	PermanentClustersDeletion(ctx context.Context, olderThan strfmt.DateTime, objectHandler s3wrapper.API) error
+	DeregisterClusterInternal(ctx context.Context, params installer.DeregisterClusterParams) error
+}
+
+func NewGarbageCollectors(
+	db *gorm.DB,
+	log logrus.FieldLogger,
+	hostApi host.API,
+	clusterApi clusterPkg.API,
+	objectHandler s3wrapper.API,
+	leaderElector leader.Leader,
+) *garbageCollector {
+	return &garbageCollector{
+		db:            db,
+		log:           log,
+		hostApi:       hostApi,
+		clusterApi:    clusterApi,
+		objectHandler: objectHandler,
+		leaderElector: leaderElector,
+	}
+}
+
+type garbageCollector struct {
+	Config
+	db            *gorm.DB
+	log           logrus.FieldLogger
+	hostApi       host.API
+	clusterApi    clusterPkg.API
+	objectHandler s3wrapper.API
+	leaderElector leader.Leader
+}
+
+func (g garbageCollector) DeregisterInactiveClusters() {
```

Comment:
should probably handled only by the leader, no need to run it on all the instances.